src/ConfigMapAndSecrets.ts

```diff
@@ -0,0 +1,231 @@
+	});
+	extension.hideStatusLine();
+}
+
+async function createSecretFromUri(uri:vscode.Uri) {
+	await createSecretFromFilenameOrFolder(uri)
+		.then ( (output) => {
+			extension.setStatusLineMessage(`Received... ${output}`);
+		}).catch( (error) => {
+			utils.shareMessage(extension.mainOutputChannel, ("Error encountered while creating Kubernetes Secret: " + error));
+	});
+	extension.hideStatusLine();
+}
+
+function validateName(text : string) {
+	return !validNameRegex.test(text) ? 'Name must be at least two characters long, start with a letter, and only include a-z, A-Z, and hyphens' : null;	
+}
+
+async function createConfigMapFromFilenameOrFolder(uri:vscode.Uri) : Promise<any> {
+	return new Promise( async (resolve, reject) => {
+		const kubeIsReady = await isKubernetesAvailable();
+		if (kubeIsReady && kubeIsReady === true) {
+			await vscode.window.showInputBox({
+				prompt: 'Kubernetes ConfigMap Name',
+				placeHolder: 'Provide the unique identifier for this Kubernetes ConfigMap.',
+				value: 'new-configmap',
+				ignoreFocusOut: true,
+				validateInput: validateName
+			}).then( (configMapName) => {
+				if (configMapName) {
+					createConfigMap(configMapName, uri)
+						.then( (output) => {
+							vscode.window.showInformationMessage(`Successfully created new Kubernetes ConfigMap named "${configMapName}"`);
+							resolve(output);
+						})
+						.catch(err => {
+							vscode.window.showErrorMessage(`Problem encountered creating new Kubernetes ConfigMap named "${configMapName}": ${err}`);
+							reject(err);
+						});
+				} else {
+					reject("No name given for Kubernetes ConfigMap.");
+				}
+			});
+		} else {
+			vscode.window.showInformationMessage(`Kubernetes extensions still coming up, please wait a moment and try again.`);
+		}
+	}); 
+}
+
+async function createSecretFromFilenameOrFolder(uri:vscode.Uri) : Promise<any> {
+	return new Promise( async (resolve, reject) => {
+		const kubeIsReady = await isKubernetesAvailable();
+		if (kubeIsReady && kubeIsReady === true) {
+			let secretName = await vscode.window.showInputBox({
+				prompt: 'Kubernetes Secret Name',
+				placeHolder: 'Provide the unique identifier for this Kubernetes secret.',
+				value: 'new-secret',
+				validateInput: validateName
+			});
+			if (secretName) {
+				createSecret(secretName, uri)
+					.then( (output) => {
+						vscode.window.showInformationMessage(`Successfully created new Kubernetes Secret named "${secretName}"`);
+						resolve(output);
+					})
+					.catch(err => {
+						vscode.window.showErrorMessage(`Problem encountered creating new Kubernetes Secret named "${secretName}": ${err}`);
+						reject(err);
+					});
+			} else {
+				reject("No name given for secret.");
+			}
+		} else {
+			vscode.window.showInformationMessage(`Kubernetes extensions still coming up, please wait a moment and try again.`);
+		}
+	}); 
+}
+
+async function isKubernetesAvailable(): Promise<any> {
```

Comment:
You are returning a boolean value always. Wouldn't it be better to specify Boolean instead of Any?